### PR TITLE
Decode filename, thumbnail and full image URLs in FrameAPI:retreiveThumbnails

### PR DESCRIPTION
**Pull request by LinnierGames**
### What's new
- Created PhotoData

### What has changed
- Updated FrameAPI:retrieveThumbnails to return PhotoData

### CODE REVIEW

**Review by LinnierGames — COMMENTED** [review]
Self-reviewed

**Comment by LinnierGames** [comment]
[Pass more info in photos/thumbnails](https://trello.com/c/8kMPXTR0/61-pass-more-info-in-photos-thumbnails)

File: PicPi/FailableDecodable.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
+//
+//  FailableDecodable.swift
+//  PicPi
+//
+//  Created by Erick Sanchez on 12/29/20.
+//
+
+import Foundation
+
+struct FailableDecodable<Base: Decodable>: Decodable {
```
Comment (RawiSoft) [inline PicPi/FailableDecodable.swift:10]: Nice!

File: PicPi/FrameAPI.swift [hunk 2]
```diff
@@ -12,6 +12,12 @@ enum FrameAPIErrors: Error {
   case dataMalformed
 }
 
+struct PhotoData: Decodable {
```
Comment (RawiSoft) [inline PicPi/FrameAPI.swift]: wdyt of changing it to photoInfo?

File: PicPi/FrameAPI.swift [hunk 3]
```diff
@@ -37,8 +43,8 @@ enum FrameAPIErrors: Error {
 protocol FrameAPI {
   /// Upload the selected photo with filename
   func upload(photoData: Data, filename: String) -> Promise<Void>
-  /// Get the list of URLs of the thumbnails stored on the Pi
-  func retrieveThumbnails() -> Promise<[URL]>
+  /// Get the list of photos and their URLs of thumbnails or full images stored on the Pi
+  func retrievePhotos() -> Promise<[PhotoData]>
```
Comment (RawiSoft) [inline PicPi/FrameAPI.swift]: wdyt of changing it to retrievePhotosInfo() ?
that will make it more readable :)
retrievePhotos()  will give the impression of retrieving the actual photo

File: PicPi/FrameAPIImpl.swift [hunk 4]
```diff
@@ -78,32 +78,27 @@ private class FrameAPIImpl: FrameAPI {
     return promise
   }
   
-  func retrieveThumbnails() -> Promise<[URL]> {
-    let promise = Promise<[URL]>.pending()
-    /// requesting the thumbnails of all the photos on the PI or Mac
-    /// result will be the URL of the thumbnail
+  
+  func retrievePhotos() -> Promise<[PhotoData]> {
```
Comment (RawiSoft) [inline PicPi/FrameAPIImpl.swift]: same as above

**Review by RawiSoft — CHANGES_REQUESTED** [review]

File: PicPi/FailableDecodable.swift (hunk 1 above)
Comment (LinnierGames) [reply PicPi/FailableDecodable.swift:10]: thanks! I spet like 2 hours trying to figure out how to decode an array of things and filter out the ones that don't work D: and this is the little result from all of that haha

File: PicPi/FrameAPI.swift (hunk 3 above)
Comment (LinnierGames) [reply PicPi/FrameAPI.swift]: sgtm 😁

File: PicPi/FrameAPI.swift (hunk 2 above)
Comment (LinnierGames) [reply PicPi/FrameAPI.swift]: sgtm wha about PhotosInfoData? the data part is so we know this is a model from json aka the HTTP API

**Pull request merged**